R/onload.R

```diff
@@ -1,6 +1,27 @@
+#' @importFrom cachem cache_disk
+#'
+#' @importFrom memoise memoise
 .onLoad <- function(libname, pkgname) {
-    # a <- Sys.getenv("DATAVERSE_SERVER")
-    # if(a == "") {
-    #     Sys.setenv("DATAVERSE_SERVER" = "dataverse.harvard.edu")
-    # }
+  # a <- Sys.getenv("DATAVERSE_SERVER")
+  # if(a == "") {
+  #     Sys.setenv("DATAVERSE_SERVER" = "dataverse.harvard.edu")
+  # }
+
+  ## implement API disk cache via 'memoise'
+  cache_directory <- file.path(
+    tools::R_user_dir(pkgname, "cache"),
+    "api_cache"
+  )
+  get <- api_get_impl
+  if (!dir.exists(cache_directory)) {
+    status <- dir.create(cache_directory, recursive = TRUE)
+    if (!status)
+      warning("'dataverse' failed to create API cache")
+  }
+  if (dir.exists(cache_directory)) {
+    # disk cache with max age 30 days
```

Comment:
Does this mean that after 30 days, the cache effectively disappears? We can document this somewhere.

Reply:
items in the cache older than 30 days disappear, but items less than 30 days remain. I can update the documentation a bit; I wasn't sure what the strategy should be -- never, x days, to keep below y MB, etc.